xbmc/guilib/GUIIncludes.cpp

```diff
@@ -530,6 +539,8 @@ const INFO::CSkinVariableString* CGUIIncludes::CreateSkinVariable(const std::str
 {
   std::map<std::string, TiXmlElement>::const_iterator it = m_skinvariables.find(name);
   if (it != m_skinvariables.end())
-    return INFO::CSkinVariable::CreateFromXML(it->second, context);
+    return INFO::CSkinVariable::CreateFromXML(it->second, context, [&](const std::string &expression) -> std::string {
```

Comment:
As expressions may not change until we reload the skin, we could first load expressions and resolve them directly for variables while we load them at https://github.com/xbmc/xbmc/blob/master/xbmc/guilib/GUIIncludes.cpp#L203

This way we would save some round trips here, or won't we?